Sources/SwiftSyntax/Identifier.swift

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// An abstraction for sanitized values on a token.
+public struct Identifier: Equatable, Sendable {
+  /// The sanitized `text` of a token.
+  public let name: String
+
+  public init(_ token: TokenSyntax) {
+    self.name = sanitizing(token.text)
+  }
+}
+
+private func sanitizing(_ name: String) -> String {
+  guard name.contains("`") else {
+    return name
+  }
+
+  return name.trimmingCharacters(in: CharacterSet(charactersIn: "`"))
+}
```

Comment:
It’s only called once so maybe it would be easier to just put the code directly into the initializer.